drivers/overlay/ovmanager/ovmanager.go

```diff
@@ -137,8 +141,8 @@ func (d *driver) NetworkFree(id string) error {
 	}
 
 	d.Lock()
+	defer d.Unlock()
```

Comment:
actual implementation locks and unlocks only at particular places. And the current change locks at the beginning of the function and  unlocks when the function exits. Like you mentioned , will there be any issue for other threads who will be waiting little longer period than it used to be ?  Just trying to see if there will be any trade off in performance or some other issue.

Reply:
There could be a tradeoff in performance here.   The only extra operation that gets protected by the lock is network.releaseVxlanID() which does two things:
1. locks itself temporarily to build a coherent list of VNIs to free
2. invokes the idm.Release() method for each VNI

The first would be protected against concurrency by the lock anyway.   The second, however, invokes the IDM which uses the bitseq package which can end up writing to store.  That last bit is the only long blocking operation I can think of here.

Even so, this is an "ovmanager.driver" instance and the only operations it locks on are creation and deletion.  I'd assume that our desired rate for creation/deletion would be measured in the order of thousands per second, in which case I highly doubt a serial write-per-op would hurt us.

But (full transparency), I chafe at bad concurrency patterns and want to make it easy to reason about the correctness of the code.  As I said in the commit log for this section:  I doubt this race is serious.